components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -69,62 +72,70 @@ const hasDecimalsOverflow = (
   return decimalPlaces !== null && decimalPlaces > 0
 }
 
+const getTokenFromParam = (
+  contractOrSymbol: string,
+  selectedNetwork: BraveWallet.NetworkInfo,
+  tokenList: BraveWallet.BlockchainToken[]
+) => {
+  return tokenList.find(
+    (token) =>
+      (token.chainId === selectedNetwork.chainId &&
+        token.contractAddress.toLowerCase() ===
+          contractOrSymbol.toLowerCase()) ||
+      (token.chainId === selectedNetwork.chainId &&
+        token.contractAddress === '' &&
+        token.symbol.toLowerCase() === contractOrSymbol.toLowerCase())
+  )
+}
+
 export const useSwap = () => {
```

Comment:
I have made big changes to this file to make the hook cross-chain, so a lot of of the routing work would need a re-implementation with my changes.

Secondly, with the Composer UI, we need these query params:
- `fromAccountId` (unique id)
- `toAccountId` (unique id)
- `fromToken` (contract address or symbol)
- `toToken` (contract address or symbol)
- `fromChainId`
- `toChainId`
- `fromCoin`
- `toCoin`

Reply:
Note that we'll initially only allow `fromChainId === toChainId`, `fromCoin === toCoin` and `fromAccountId === toAccountId`, but those restrictions would be removed very soon.